File: 6KYU/Thinking & Testing: A * B/solution.go
package kata

func SumDigits(a int) int {
	res := 0
	for {
		if a == 0 {
			break
		}
		d := a % 10
		a /= 10
		res += d
	}
	return res
}

func TestIt(a, b int) int {
	return SumDigits(a) * SumDigits(b)
}

File: 6KYU/Thinking & Testing: A * B/solution.rs
fn sum_digits(n: u64) -> u64 {
    let mut sum = 0;
    let mut n = n;
    
    while n != 0 {
        let d = n % 10;
        sum += d;
        n /= 10;
    }
    
    return sum;
}

fn test_it(a: u64, b: u64) -> u64 {
    return sum_digits(a) * sum_digits(b);
}
